### PR TITLE
fix(hip,aiter): route single-prefill bf16/hd128 to ASM v3 batch mode

### PR DESCRIPTION
**Pull request by demandal25**
## Summary

Route `SinglePrefillWithKVCacheDispatched` to AITER's ASM v3 batch-mode pipeline when the runtime traits match the shapes AITER ships as precompiled `.co` files. For ineligible shapes the call falls back to the existing group-mode CK Tile path; no behavior change in that branch.

### What changed

- **`include/flashinfer/attention/aiter/single_prefill.cuh`** — added `AiterAsmV3HdimSupported` and `AiterAsmV3Eligible` constexpr predicates mirroring the guard in `aiter::fmha_fwd_v3` (`mha_fwd.cpp:188`). The kernel-launch path now sets `args.use_asm_v3 = AiterAsmV3Eligible(...)` and `args.is_group_mode = !args.use_asm_v3`. Seqstart pointers become nullable on the ASM v3 branch.
- **`flashinfer/csrc_rocm/single_prefill_aiter.cu`** — caller skips the `get_seqstart_ptrs` cache lookup (mutex + hash + at::Tensor data_ptr) when the ASM v3 path will be taken.

### Architecture / design notes

| Trait | ASM v3 path | CK Tile fallback |
|---|---|---|
| dtype | bf16 only | fp16 + bf16 |
| (hdim_q, hdim_v) | (128, 128) or (192, 128) | any |
| logits soft cap | unsupported | supported |
| sliding window | unsupported | supported |
| arch | gfx942 / gfx950 | gfx942 / gfx950 |
| seqstart pointers | nullptr (batch=1) | required ([0, len]) |
| `is_group_mode` | `false` | `true` |

The eligibility predicate is intentionally computed at both the `.cu` call site (to gate the seqstart cache lookup) and inside the `.cuh` template (to keep the framework-agnostic header self-contained). Both use the same `AiterAsmV3Eligible(HEAD_DIM_QK, HEAD_DIM_VO, dtype_enum, has_logits_cap, window_left)` so the two sites can't drift.

The CK Tile fallback must stay group-mode because the variant `.so` we dlopen is JIT-built with group-mode CK instances only — `is_group_mode=false` on that path would hit a kernel-not-found.

## Test plan

- [x] `pytest tests/rocm_tests/test_single_prefill_kernels_hip.py -k aiter --reruns 1` — **820 passed**, 1104 skipped (covers fp16 + bf16, hd ∈ {64, 128, 256}, causal/non-causal, LSE on/off — exercises both the new ASM v3 branch and the unchanged CK fallback)
- [x] `pytest tests/rocm_tests/test_single_prefill_kernels_hip.py --reruns 1` — **2456 passed**, 1392 skipped
- [x] `pre-commit run` on modified files

### CODE REVIEW

File: include/flashinfer/attention/aiter/single_prefill.cuh [hunk 1]
```diff
@@ -21,8 +21,31 @@ inline constexpr int32_t kAiterMaskTopLeft = 1;  // standard causal (qo_len == k
...
     2;  // prefill-with-history causal (kv_len > qo_len)
 
+// AITER's ASM v3 fast path is restricted to the (hdim_q, hdim_v) pairs that ship
+// as precompiled .co files in aiter_meta/hsa/gfx9{42,50}/fmha_v3_fwd/. Any other
+// shape forces a fallback to fmha_fwd_ck, which in the variant .so we dlopen is
+// JIT-built group-mode only — so callers must keep is_group_mode=true unless
+// ASM v3 will actually be selected.
+inline constexpr bool AiterAsmV3HdimSupported(uint32_t hdim_q, uint32_t hdim_v) {
+  return (hdim_q == 128 && hdim_v == 128) || (hdim_q == 192 && hdim_v == 128);
+}
```
[the remark this answers is not in the document]
Comment (demandal25) [reply include/flashinfer/attention/aiter/single_prefill.cuh:33]: Good catch — fixed in 717b4a55. (192, 128) is unreachable through this entry point because of the HEAD_DIM_QK == HEAD_DIM_VO static_assert; dropped that arm from AiterAsmV3HdimSupported and added a comment explaining the constraint.

File: include/flashinfer/attention/aiter/single_prefill.cuh [hunk 2]
```diff
@@ -66,9 +91,11 @@ hipError_t SinglePrefillWithKVCacheDispatched(Params const& params, bool causal,
...
   args.lse_ptr = static_cast<void*>(params.lse);
 
-  // Group mode with seqstart arrays [0, seqlen] encodes a single-sequence batch.
-  args.seqstart_q_ptr = static_cast<const void*>(cu_seqlens_q);
-  args.seqstart_k_ptr = static_cast<const void*>(cu_seqlens_k);
+  // Group mode encodes batch=1 as seqstart arrays [0, seqlen] on device.
+  // Batch mode addresses tensors via stride/nhead_stride only — batch_stride_*
+  // are unused at batch=1 and left at their default 0.
+  args.seqstart_q_ptr = args.use_asm_v3 ? nullptr : static_cast<const void*>(cu_seqlens_q);
+  args.seqstart_k_ptr = args.use_asm_v3 ? nullptr : static_cast<const void*>(cu_seqlens_k);
```
[the remark this answers is not in the document]
Comment (demandal25) [reply include/flashinfer/attention/aiter/single_prefill.cuh:106]: Agreed — fixed in 717b4a55. Added a defensive nullptr check that returns hipErrorInvalidValue when group mode is selected but seqstart pointers are missing, so caller/callee predicate drift surfaces here rather than as a HIP memory fault inside AITER.
